src/Microsoft.AspNet.Server.Kestrel/Networking/UvWriteReq2.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.AspNet.Server.Kestrel.Infrastructure;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNet.Server.Kestrel.Networking
+{
+    /// <summary>
+    /// Summary description for UvWriteRequest2
+    /// </summary>
+    public class UvWriteReq2 : UvRequest
+    {
+        private readonly static Libuv.uv_write_cb _uv_write2_cb = (IntPtr ptr, int status) => UvWrite2Cb(ptr, status);
+
+        private IntPtr _bufs;
+
+        private Action<UvWriteReq2, int, Exception, object> _callback;
+        private object _state;
+        private const int BUFFER_COUNT = 1;
+
+        private GCHandle _pinUvWrite2Req;
+        private GCHandle _pinBuffer;
+        private bool _bufferIsPinned;
+
+        public UvWriteReq2(IKestrelTrace logger) : base(logger)
+        {
+        }
+
+        public void Init(UvLoopHandle loop)
+        {
+            var requestSize = loop.Libuv.req_size(Libuv.RequestType.WRITE);
+            var bufferSize = Marshal.SizeOf<Libuv.uv_buf_t>() * BUFFER_COUNT;
+            CreateMemory(
+                loop.Libuv,
+                loop.ThreadId,
+                requestSize + bufferSize);
+            _bufs = handle + requestSize;
+        }
+
+        public unsafe void Write2(
+            UvStreamHandle handle,
+            ArraySegment<byte> buf,
```

Comment:
Made change for this but can't test due to https://github.com/aspnet/Home/issues/1207